### DISCUSSION TITLE
Please improve the highlighting rules

### DISCUSSION THREAD

**Discussion by MCcber** [icsharpcode/AvalonEdit#419] [Ideas]
I am developing an AutoCompletion function for a command line language. I currently encounter a situation where the command radical and the parameter of the command use the same word. The current highlighting rules cannot distinguish whether it is the parameter or the radical and then use the correct coloring.The code below is an example.

```
<Span color="literalBlue" begin="ambient|block|hostile|master|music|neutral|player|record|voice|weather" end="\s" />
```

```
<Span color="radicalPink" begin="weather" end="\s" />
```

So in this case, only one coloring of the word weather will be applied.

So I would like to ask if it is possible to develop two new attributes to determine the characters that need to appear at the beginning and end, and then use the current begin and end as attributes to perform coloring?

**Comment by ltrzesniewski** [comment]
FWIW, you can apply multiple `HighlightingColorizer` instances to the line transformers if it's easier for you. You may also use `RichTextColorizer ` or implement your own logic.

But for your specific case, I'd say you can use a `<Rule>` with a regex containing a lookbehind such as `\bweather\b(?&lt;=^\s*\w+)` (`(?&lt;=` is needed instead of `(?<=` since you're in an xml file).

**Reply by MCcber** [reply]
Thanks bro, I will try it
